File: P1S3/src/AdminFiltros.cpp
#include "AdminFiltros.h"

AdminFiltros::AdminFiltros(): cadenaFiltros(){}

void AdminFiltros::setFiltro(Filtro &filtro) {
	cadenaFiltros.addFiltro(filtro);
}

void AdminFiltros::setTarget(Post &post) {
	cadenaFiltros.setTarget(post);
}

void AdminFiltros::ejecutar() {
	cadenaFiltros.ejecutar();
}

File: P2/AdminFiltros.dart
import 'CadenaFiltros.dart';
import 'Post.dart';
import 'Filtro.dart';


class AdminFiltros{
  CadenaFiltros cadenaFiltros;

 void setFiltro(Filtro filtro) {
   this.cadenaFiltros.addFiltro(filtro);
  }

 void setTarget(Post post) {
   this.cadenaFiltros.setTarget(post);
  }
  
  List<Filtro> getFiltros(){
    return cadenaFiltros.getFiltros();
  }

  AdminFiltros(){
   this.cadenaFiltros = new CadenaFiltros(null);
  }

 void ejecutar() {
   this.cadenaFiltros.ejecutar();
  }
}